diag_manager/diag_manager.F90

```diff
@@ -436,13 +436,20 @@ INTEGER FUNCTION register_diag_field_array(module_name, field_name, axes, init_t
     INTEGER,          OPTIONAL, INTENT(in) :: area          !< Id of the area field
     INTEGER,          OPTIONAL, INTENT(in) :: volume        !< Id of the volume field
     CHARACTER(len=*), OPTIONAL, INTENT(in) :: realm         !< String to set as the modeling_realm attribute
+    logical                                :: allow_log
 
     if (use_modern_diag) then
+      ! check if logging registered fields, uses arg and nml flag
+      IF ( PRESENT(do_not_log) ) THEN
+         allow_log = .NOT.do_not_log
+      ELSE
+         allow_log = .TRUE.
+      END IF
       register_diag_field_array = fms_diag_object%fms_register_diag_field_array( &
        & module_name, field_name, axes, init_time, long_name=long_name, &
        & units=units, missing_value=missing_value, var_range=range, mask_variant=mask_variant, &
-       & standard_name=standard_name, verbose=verbose, do_not_log=do_not_log, err_msg=err_msg, &
-       & interp_method=interp_method, tile_count=tile_count, area=area, volume=volume, realm=realm)
+       & standard_name=standard_name, verbose=verbose, do_not_log=.not.(do_diag_field_log.AND.allow_log), &
```

Comment:
This is adding unneeded complexity.  do_not_log is optional, so if it's not present, then it won't be passed on because it's optional in the next routine (register_diag_field_array) as well.  

You should move this logic to where it's actually used instead of now actually passing something to do_not_log.

Reply:
The logic basically prints the log if either the nml flag or do_not_log is set to output , I added it here cause I needed the nml flag and this was the last place where it would be accessible.

I'm gonna change it around anyway so I might take out the do_not_log from the object's register routine if it ends up being useless (which seems likely)